### PR TITLE
Add flows

### PR DESCRIPTION
**Pull request by katlyn333**
We added flow support to the salesforce gem. A flow is just a configurable API endpoint that you can ping with parameters and it kicks off some sort of process in Salesforce.

### CODE REVIEW

**Comment by katlyn333** [comment]
@cantino Could you please review? We will probably submit a pull request to their repo so we want to make sure its up to snuff.

File: README.md [hunk 1]
```diff
@@ -240,6 +240,23 @@ me.follow(you)                                            #=> start following a
...
 
+## Running a Salesforce Flow
+You can start Salesforce flows.
+
+```ruby
+flow = Databasedotcom::Flow.new(client, flow_name, flow_paramaters)
+flow.run
+```
+
+It triggers a post to ```/services/data/v32.0/actions/custom/flow/#{@flow_name}``` with the passed in paramaters.
```
Comment (cantino) [inline README.md]: "Executing `run` will trigger a post to"

File: README.md [hunk 2]
```diff
@@ -240,6 +240,23 @@ me.follow(you)                                            #=> start following a
...
+
+Configure the API version by passing it in as an option.
```
Comment (cantino) [inline README.md]: You can configure the API version with the `:version` option:

File: lib/databasedotcom/flow.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+module Databasedotcom
+  # A Salesforce API endpoint that can trigger predefinted flows in salesforce to kick off salesforce side processes.
+  # A flow can be triggered by its unique developer name and passed a hash of input variables.
```
Comment (cantino) [inline lib/databasedotcom/flow.rb]: `# A flow can be triggered by its unique name and can be passed a hash of input variables.`

File: lib/databasedotcom/flow.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    def run
+      params_hash = {inputs: [@params]}
+      @client.http_post(flow_url, params_hash.to_json)
+    end
+
+    protected
+
+    def flow_url
+      "/services/data/v#{@version}/actions/custom/flow/#{@flow_name}"
```
Comment (cantino) [inline lib/databasedotcom/flow.rb:21]: Do we need to URL-escape the `@flow_name`?  What characters do they allow?

Comment (katlyn333) [inline lib/databasedotcom/flow.rb:21]: Hmm good point. I took a look at the docs and flow names cannot contain spaces in salesforce and are automatically separated by underscores. The is considered the API name for the flow so they handle validation on creation in their UI.

**Comment by katlyn333** [comment]
Ok changes made

**Comment by cantino** [comment]
Seems fine to me!

**Pull request merged**